virtwho.py

```diff
@@ -175,50 +240,67 @@ def run(self):
             return
 
         result = {}
+        report = None
         while not self.terminate_event.is_set():
             # Wait for incoming report from virt backend
             try:
-                report = self.queue.get(block=True, timeout=None)
+                report = self.queue.get_nowait()
```

Comment:
I don't think it's necessary for each Job to have it's own timeout. Just create list of Jobs to be checked next time we're checking them.